src/allmydata/scripts/tahoe_put.py

```diff
@@ -80,8 +84,13 @@ def put(options):
         # Content-Length field. So we currently must copy it.
         if verbosity > 0:
             print("waiting for file data on stdin..", file=stderr)
-        data = stdin.read()
-        infileobj = StringIO(data)
+        # We're uploading arbitrary files, so this had better be bytes:
+        if PY2:
+            stdinb = stdin
+        else:
+            stdinb = stdin.buffer
+        data = stdinb.read()
+        infileobj = BytesIO(data)
```

Comment:
Just a drive-by comment to lament the fact we're putting all this in memory :( (but, so did the old code so no action for this PR)